src/purchases.ts

```diff
@@ -218,15 +245,29 @@ export default class Purchases {
       throw new Error("appUserID needs to be a string");
     }
 
+    let purchasesCompletedByToUse: PURCHASES_ARE_COMPLETED_BY_TYPE;
+    let storeKitVersionToUse = storeKitVersion;
+
+    if (Purchases.isPurchasesAreCompletedByMyApp(purchasesAreCompletedBy)) {
+      purchasesCompletedByToUse = PURCHASES_ARE_COMPLETED_BY_TYPE.MY_APP;
+      storeKitVersionToUse = (
+        purchasesAreCompletedBy as PurchasesAreCompletedByMyApp
```

Comment:
The typecasting here feels a bit icky to me, but we know it'll work due to the `isPurchasesAreCompletedByMyApp()` call

Reply:
We could try to use typeguards I think: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards. I don't think we've used those before but seems like it would make the code nicer. Lmk if I can help with this!